persistence/src/commonTest/kotlin/com/wire/kalium/persistence/dao/ConversationDAOTest.kt

```diff
@@ -497,24 +496,80 @@ class ConversationDAOTest : BaseDatabaseTest() {
         )
 
         userDAO.insertUser(user1)
+        val readMessagesCount = 3
+        val unreadMessagesCount = 2
 
         val message = buildList {
             // add 9 Message before the lastReadDate
-            repeat(9) {
+            repeat(readMessagesCount) {
                 add(
                     newRegularMessageEntity(
-                        id = it.toString(),
+                        id = "read$it",
                         date = "2000-01-01T11:0$it:00.000Z",
                         conversationId = conversationId,
                         senderUserId = user1.id,
                     )
                 )
             }
             // add 9 Message past the lastReadDate
-            repeat(9) {
+            repeat(unreadMessagesCount) {
+                add(
+                    newRegularMessageEntity(
+                        id = "unread$it",
+                        date = "2000-01-01T13:0$it:00.000Z",
+                        conversationId = conversationId,
+                        senderUserId = user1.id,
+                    )
+                )
+            }
+        }
+
+        messageDAO.insertMessages(message)
+
+        launch(UnconfinedTestDispatcher(testScheduler)) {
+            // when
+            conversationDAO.observeGetConversationByQualifiedID(conversationId).test {
+                val conversation = awaitItem()
+                assertNotNull(conversation)
+                // then
+                assertEquals(unreadMessagesCount, conversation.unreadContentCountEntity.values.sum())
+            }
+        }
+    }
+
+    @Test
+    fun givenDifferentUnreadMessageContentTypes_WhenGettingUnreadMessageCount_ThenSystemMessagesShouldBeNotCounted() = runTest {
```

Comment:
It would be nice to have the test cases for the specific message types.

Like making sure that the count is correct when inserting text, asset, ping messages, etc. instead of just the sum of everything.